llvm/test/Transforms/SimplifyCFG/fold-i1.ll

```diff
@@ -0,0 +1,64 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 4
+; RUN: opt < %s -S -passes='simplifycfg<switch-to-lookup>' | FileCheck %s
+
+target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
+
+define { i1, i8 } @f(i8 noundef %character) unnamed_addr {
+; CHECK-LABEL: define { i1, i8 } @f(
+; CHECK-SAME: i8 noundef [[CHARACTER:%.*]]) unnamed_addr {
+; CHECK-NEXT:  start:
+; CHECK-NEXT:    [[SWITCH_TABLEIDX:%.*]] = sub i8 [[CHARACTER]], 48
+; CHECK-NEXT:    [[TMP2:%.*]] = icmp ult i8 [[SWITCH_TABLEIDX]], 9
+; CHECK-NEXT:    [[_0_SROA_11_0:%.*]] = select i1 [[TMP2]], i8 [[SWITCH_TABLEIDX]], i8 undef
+; CHECK-NEXT:    [[_0_SROA_0_0:%.*]] = select i1 [[TMP2]], i1 true, i1 false
```

Comment:
It would be good to also test the case where we actually build a lookup table for the booleans. Need some pattern for that which doesn't reduce down to a simple select.

If I'm reading the code right, then what we'd end up doing is generate an array of i1 constants and then generate i1 loads -- the former is okay-ish, but the latter is a big no-no. We'd instead want to generate a table of i8s, generate i8 loads and truncate them to i1.

An alternative would be to not allow table generation, but instead relax the "AllTablesFitInRegister" condition so that we always have either a legal integer width or the lookup mask fitting in a register, without requiring that *all* of them fit in a register.